README


This archive contains the distribution Carp-Always-Color,
version 0.07:

  Carp::Always, but with color

This software is copyright (c) 2013 by Jesse Luehrs.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.




Changes
Revision history for Carp::Always::Color

0.07  2013-01-02
      - Carp::Always::Color now no longer falls back to HTML coloring by
        default (you can get this behavior by using Carp::Always::Color::HTML
        explicitly, if desired). Carp::Always::Color itself now just leaves the
        output uncolored for non-terminal outputs.

0.06  2012-03-08
      - fix some test stuff

0.05  2011-09-04
      - un-todo some stuff (and fix a broken test) for the fixes in
        Carp::Always 0.10

0.04  2010-06-14
      - add strict and warnings
      - convert to more dzil stuff

0.03  2010-04-30
      - Sigh, stupid packaging (remove stray test.pl from dist)

0.02  2010-04-30
      - Fix some bugs dealing with evals and non-string exceptions

0.01  2010-04-30
      - Initial release


test.pl
#!/usr/bin/env perl
use strict;
use warnings;

use Carp::Always::Color::Term;
eval { die "foo" };
die $@;


lib/Carp/Always/Color.pm
package Carp::Always::Color;
BEGIN {
  $Carp::Always::Color::AUTHORITY = 'cpan:DOY';
}
{
  $Carp::Always::Color::VERSION = '0.07';
}
use strict;
use warnings;
# ABSTRACT: Carp::Always, but with color

BEGIN {
    if (-t *STDERR) {
        require Carp::Always::Color::Term;
    }
}


1;

__END__

=pod

=head1 NAME

Carp::Always::Color - Carp::Always, but with color

=head1 VERSION

version 0.07

=head1 SYNOPSIS

  use Carp::Always::Color;

or

  perl -MCarp::Always::Color -e'sub foo { die "foo" } foo()'

=head1 DESCRIPTION

Stack traces are hard to read when the messages wrap, because it's hard to tell
when one message ends and the next message starts. This just colors the first
line of each stacktrace, based on whether it's a warning or an error. If
messages are being sent to a terminal, it colors them with terminal escape
codes. If you want to force this behavior, you can use
L<Carp::Always::Color::Term> instead, which will always add terminal escape
codes, even when the messages are being sent to something that doesn't look
like a terminal. L<Carp::Always::Color::HTML> also exists, to add HTML color
markup to the messages instead of terminal color codes.

=head1 BUGS

No known bugs.

Please report any bugs through RT: email
C<bug-carp-always-color at rt.cpan.org>, or browse to
L<http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Carp-Always-Color>.

=head1 SEE ALSO

L<Carp::Always>

=head1 SUPPORT

You can find this documentation for this module with the perldoc command.

    perldoc Carp::Always::Color

You can also look for information at:

=over 4

=item * AnnoCPAN: Annotated CPAN documentation

L<http://annocpan.org/dist/Carp-Always-Color>

=item * CPAN Ratings

L<http://cpanratings.perl.org/d/Carp-Always-Color>

=item * RT: CPAN's request tracker

L<http://rt.cpan.org/NoAuth/Bugs.html?Dist=Carp-Always-Color>

=item * Search CPAN

L<http://search.cpan.org/dist/Carp-Always-Color>

=back

=head1 AUTHOR

Jesse Luehrs <doy at tozt dot net>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2013 by Jesse Luehrs.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


lib/Carp/Always/Color/HTML.pm
package Carp::Always::Color::HTML;
BEGIN {
  $Carp::Always::Color::HTML::AUTHORITY = 'cpan:DOY';
}
{
  $Carp::Always::Color::HTML::VERSION = '0.07';
}
use strict;
use warnings;
use Carp::Always 0.10;
# ABSTRACT: Carp::Always, but with HTML color


BEGIN { $Carp::Internal{(__PACKAGE__)}++ }

sub _die {
    die @_ if ref($_[0]);
    eval { Carp::Always::_die(@_) };
    my $err = $@;
    $err =~ s/(.*)( at .*? line .*?$)/<span style="color:#800">$1<\/span>$2/m;
    die $err;
}

sub _warn {
    my @warning;
    {
        local $SIG{__WARN__} = sub { @warning = @_ };
        Carp::Always::_warn(@_);
    }
    $warning[0] =~ s/(.*)( at .*? line .*?$)/<span style="color:#880">$1<\/span>$2/m;
    warn @warning;
}

my %OLD_SIG;
BEGIN {
    @OLD_SIG{qw(__DIE__ __WARN__)} = @SIG{qw(__DIE__ __WARN__)};
    $SIG{__DIE__} = \&_die;
    $SIG{__WARN__} = \&_warn;
}

END {
    @SIG{qw(__DIE__ __WARN__)} = @OLD_SIG{qw(__DIE__ __WARN__)};
}

1;

__END__

=pod

=head1 NAME

Carp::Always::Color::HTML - Carp::Always, but with HTML color

=head1 VERSION

version 0.07

=head1 SYNOPSIS

  use Carp::Always::Color::HTML;

or

  perl -MCarp::Always::Color::HTML -e'sub foo { die "foo" } foo()'

=head1 DESCRIPTION

Like L<Carp::Always::Color>, but forces HTML coloring, regardless of where
STDERR is pointing to.

=head1 AUTHOR

Jesse Luehrs <doy at tozt dot net>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2013 by Jesse Luehrs.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


lib/Carp/Always/Color/Term.pm
package Carp::Always::Color::Term;
BEGIN {
  $Carp::Always::Color::Term::AUTHORITY = 'cpan:DOY';
}
{
  $Carp::Always::Color::Term::VERSION = '0.07';
}
use strict;
use warnings;
use Carp::Always 0.10;
# ABSTRACT: Carp::Always, but with terminal color


BEGIN { $Carp::Internal{(__PACKAGE__)}++ }

sub _die {
    die @_ if ref($_[0]);
    eval { Carp::Always::_die(@_) };
    my $err = $@;
    $err =~ s/(.*)( at .*? line .*?$)/\e[31m$1\e[m$2/m;
    die $err;
}

sub _warn {
    my @warning;
    {
        local $SIG{__WARN__} = sub { @warning = @_ };
        Carp::Always::_warn(@_);
    }
    $warning[0] =~ s/(.*)( at .*? line .*?$)/\e[33m$1\e[m$2/m;
    warn @warning;
}

my %OLD_SIG;
BEGIN {
    @OLD_SIG{qw(__DIE__ __WARN__)} = @SIG{qw(__DIE__ __WARN__)};
    $SIG{__DIE__} = \&_die;
    $SIG{__WARN__} = \&_warn;
}

END {
    @SIG{qw(__DIE__ __WARN__)} = @OLD_SIG{qw(__DIE__ __WARN__)};
}

1;

__END__

=pod

=head1 NAME

Carp::Always::Color::Term - Carp::Always, but with terminal color

=head1 VERSION

version 0.07

=head1 SYNOPSIS

  use Carp::Always::Color::Term;

or

  perl -MCarp::Always::Color::Term -e'sub foo { die "foo" } foo()'

=head1 DESCRIPTION

Like L<Carp::Always::Color>, but forces ANSI terminal code coloring, regardless
of where STDERR is pointing to.

=head1 AUTHOR

Jesse Luehrs <doy at tozt dot net>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2013 by Jesse Luehrs.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Makefile.PL

use strict;
use warnings;



use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Carp::Always, but with color",
  "AUTHOR" => "Jesse Luehrs <doy at tozt dot net>",
  "BUILD_REQUIRES" => {
    "B" => 0,
    "File::Find" => 0,
    "File::Temp" => 0,
    "Test::More" => "0.88"
  },
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Carp-Always-Color",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Carp::Always::Color",
  "PREREQ_PM" => {
    "Carp::Always" => "0.10",
    "strict" => 0,
    "warnings" => 0
  },
  "VERSION" => "0.07",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



